server/models.py

```diff
@@ -404,6 +404,10 @@ class Batch(BaseModel):
         order_by="BatchResultTallySheet.created_at",
     )
 
+    last_edited_by_user_type = Column(String(200))
+    last_edited_by_user_key = Column(String(200))
+    last_edited_by_support_user_email = Column(String(200))
```

Comment:
I considered a number of possible schema, including foreign keys into the users / tally entry users tables, but ultimately settled on these three session fields as a simple way to identify users of various types

Reply:
Yeah you could unify all users in one table, but that's a very big change.

I'm a tad irked by the fact that `user_key` really refers to different fields in different tables depending on `user_type`. But I don't have anything better to propose, so I'm okay with it. Maybe document a little bit more what `user_key` is here?

Reply:
> One more general solution might be to model all users in one table with extra tables for data specific to each user type. That might be out of scope for this task though.

Modeling all users in one table definitely crossed my mind but of course felt out of scope

> That being said, maybe we can ask if this solution could be easily migrated to that one. I think the answer is yes, because we are storing enough info to reference the existing user record.

:100: Yeah, this was definitely a priority, being able to identify the exact user. That's why I avoided a simple string model, e.g. `last_edited_by: "Alice, Bob"`

> I'm a tad irked by the fact that `user_key` really refers to different fields in different tables depending on `user_type`. But I don't have anything better to propose, so I'm okay with it. Maybe document a little bit more what `user_key` is here?

Will add a comment! For reference, I picked these names to match the names of the fields that we store on the session:

https://github.com/votingworks/arlo/blob/551d8c4e223672ca1abfb502dba1127c3c45c5ef/server/auth/auth_helpers.py#L51-L54

I'll note one last alternative that I considered that would've avoided the vague names: separate fields for the different user types, e.g. `last_edited_by_jurisdiction_admin_email`, `last_edited_by_tally_entry_user_id`, etc. A downside to this model was that we'd have to enforce that only one of the fields was filled in at a time with application code.

Reply:
Update: Did end up switching things up, mainly to avoid an inefficient query fanout: https://github.com/votingworks/arlo/pull/1673#issuecomment-1310887267

> A downside to this model was that we'd have to enforce that only one of the fields was filled in at a time with application code

Found a way to do this without application code and with a SQL constraint instead! https://github.com/votingworks/arlo/pull/1673#discussion_r1019718841